### DISCUSSION TITLE
Let the numbers talk about product!!

### DISCUSSION THREAD

**Discussion by shelgeram07** [SigNoz/signoz#218] [Ideas]
I was going through the documentation and I did not find anything about scalability or performance numbers. Please point me to the doc if there is any data around that.

When we are talking about a product which is in APM space there are certain things people would be looking before trying out any product like limitations, stats around performance, stability and scalability of the product. 

From APM Perspective, It would be great if we can publish some stats around scalability, performance for few metrics like 

1. Number of logs per second it can ingest?
2. Any baseline numbers of read or query performance?
3. Do we have some baseline or benchmark numbers on infrastructure needed to support 'x' log writes per second?
4. What is the typical overhead of SigNoz ? I know this is heavily dependent on application but it would be good if we have some data/ stats on this as well.

This also brings me to think on what are the current performance testing practices which we follow and how can that be improved on. I would love to know more about it and share my 2 cent of knowledge to make it more scalable, reliable and performant!!

**Comment by pranay01** [comment]
Thanks @shelgeram07 for your note. 

We did benchmark the cost of running SigNoz against SaaS products ( https://signoz.io/blog/signoz-benchmarks/)  but we haven't done any detailed benchmarks on performance and scalability. You are right, this is something which would be helpful to know before starting to use SigNoz in production.

As discussed in our call, can you help us put a framework around this - since you already have extensive experience in this domain. We will soon be setting up an architecture deep dive session with core maintainers so that you have more context and we can take this further. Will keep you posted

**Comment by pranay01** [comment]
hey @shelgeram07 I hope the community call on data flow was useful for understanding the architecture in more detail. Looking forward forward to the doc from you to start more detailed discussions on how we can set the performance testing framework

**Reply by shelgeram07** [reply]
@pranay01  Sure. I am working on test strategy document and hopefully we will have it ready in couple of days for review.
